plugin/evm/vm_test.go

```diff
@@ -376,6 +377,73 @@ func issueAndAccept(t *testing.T, issuer <-chan engCommon.Message, vm *VM) snowm
 	return blk
 }
 
+func TestSubnetEVMUpgradeRequiredAtGenesis(t *testing.T) {
+	genesisTests := []struct {
+		genesisJSON string
+	}{
+		{
+			// we expect an error when subnet evm upgrade is nil in chain config
+			genesisJSON: genesisJSONPreSubnetEVM,
+		},
+		{
+			// we expect an error when subnet evm upgrade is not enabled at genesis and at a later block instead
+			genesisJSON: genesisJSONSubnetEVMLateEnablement,
+		},
+	}
+
+	for _, test := range genesisTests {
+		ctx, dbManager, genesisBytes, issuer := setupGenesis(t, test.genesisJSON)
+		vm := &VM{}
+		err := vm.Initialize(
+			context.Background(),
+			ctx,
+			dbManager,
+			genesisBytes,
+			[]byte(""),
+			[]byte(""),
+			issuer,
+			[]*engCommon.Fx{},
+			nil,
+		)
+
+		require.ErrorIs(t, err, errSubnetEVMUpgradeNotEnabled)
+	}
+}
+
+func TestOverrideSkipSubnetEVMUpgrade(t *testing.T) {
+	genesisTests := []struct {
+		genesisJSON string
+	}{
+		{
+			// we expect an error when subnet evm upgrade is nil in chain config
```

Comment:
can we update these comments or merge the 2 tests by adding `configJSON` and `expectedErr` in the test struct?

Reply:
yes